src/charm.py

```diff
@@ -195,76 +143,12 @@ def _external_url(self) -> str:
 
     @property
     def _internal_url(self) -> str:
-        scheme = "https" if self.tls_available else "http"
-        return f"{scheme}://{self.hostname}"
-
-    @property
-    def tls_available(self) -> bool:
-        """Return True if tls is enabled and the necessary certs are found."""
-        return (
-            self.cert_handler.enabled
-            and (self.cert_handler.server_cert is not None)
-            and (self.cert_handler.private_key is not None)
-            and (self.cert_handler.ca_cert is not None)
-        )
+        """Returns workload's FQDN."""
+        scheme = "http"
+        if hasattr(self, "coordinator") and self.coordinator.nginx.are_certificates_on_disk:
```

Comment:
IIRC this isn't necessary like in the worker

Reply:
its the other way around I believe. The many related PRs are getting confusing now.